src/bops-functions/bops-engine/variables/functions/decrease-variable.ts

```diff
@@ -0,0 +1,24 @@
+import { InternalMetaFunction } from "@api/bops-functions/internal-meta-function";
+
+export function decreaseVariableFunction (input : { variableName : string; value ?: number }) : unknown {
+  if(typeof this.variables[input.variableName] !== "number") {
```

Comment:
I think that using `this` is not a good practice in all circumstances for a BOps Function because it bleeds the context into a function, which can lead to unexpected side effects in a BOp. What is the reasoning behind this?

Reply:
I actually agree on this one. I have since changed the variable functions to receive a separate parameter with the variables instead of using `this`.